File: study/app/src/main/java/com/learning/study/MainActivity.java
package com.learning.study;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {
    static String value = "value";
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Intent intent = getIntent();
        System.out.println(intent.getStringExtra(value));
        if (intent.getStringExtra(value) != null)
            Toast.makeText(this, intent.getStringExtra(value), Toast.LENGTH_SHORT).show();
    }

    public void someFunc(View view) {
        TextView textView = (TextView) view;
        Toast.makeText(this, String.valueOf(textView.getText()), Toast.LENGTH_SHORT).show();
    }

    public void buttonFunc(View view) {
        Intent intent = new Intent(this, MainActivity.class);
        intent.putExtra(value, "extra!!!!!!!!!!!!!!!!!!1");
        startActivity(intent);
    }
}

File: Lab1/app/src/main/java/com/example/myapplication/MainActivity.kt
package com.example.myapplication

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.*

class MainActivity : AppCompatActivity() {
    private lateinit var button_ok : Button
    private lateinit var button_cancel : Button
    private lateinit var spinner : Spinner
    private lateinit var text : TextView

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        button_ok = findViewById(R.id.ok)
        button_cancel = findViewById(R.id.cancel)
        spinner = findViewById(R.id.spinner)
        text = findViewById(R.id.output)

        button_cancel.setOnClickListener {view ->
            text.setText("")
        }

        button_ok.setOnClickListener { view ->
            var pos: Int = spinner.getSelectedItemPosition()
            if (pos != 0)
                text.setText(spinner.getSelectedItem() as String)
            else
                Toast.makeText(getApplicationContext(), "Будь ласка виберіть мову", Toast.LENGTH_SHORT).show()
        }
    }
}